### ISSUE TITLE
BUG: double death occurs sometimes

### ISSUE THREAD

**Issue by modelarious** [Cosmic-Cryptid-Games/GrowingDepths#86]
Sometimes on death, the character will respawn, then respawn again

**Comment by modelarious** [comment]
The double death is caused by something calling the common event that triggers death twice. 
 I've observed it being called twice and resetting the player twice by using "console.log" to tell me when the common event is entered.

Easiest way I've found to reproduce this is by using the death pit in Treetops 3.

Possible fix: timer to reset the common event, so this would mean that it can't be triggered more than once within (for example) a second

other possible fix: maybe where death is handled in TMJumpAction is the issue

**Comment by modelarious** [comment]
Here's the problem code in TMJumpAction (around line 2100):

```
case actInstantKillRegion:
          var battler = this.actor()
          battler.addState(0001)
          break;
```

It gets called like a maniac when the player makes contact with the death blocks, which causes common event 1 to get triggered a bunch

**Comment by modelarious** [comment]
oooh! you could control where execution is by having a variable protect the common event and the code.. something like

```
case actInstantKillRegion:
    if ($gameVariable[1] == 1) {

        //prevents this from running a bunch of times until the 
        //common event sets game variable 1 back to the value 1
        $gameVariable[1] = 0; 
        var battler = this.actor()
        battler.addState(0001)
    }
    break;
```

and then in the common event

```
death stuff
wait 1 sec
gameVariable[1] = 1
```

**Comment by modelarious** [comment]
It seems like the new method of killing the player when they are in region 66 has brought this bug back.  All that needs to be done to fix this is to set `variable 12` to 1 when the player enters the region.

**Comment by modelarious** [comment]
Fixed this in Mariah's #159
